limitador/src/storage/keys.rs

```diff
@@ -187,6 +221,20 @@ pub mod bin {
         }
     }
 
+    pub fn key_for_counter_v2(counter: &Counter) -> Vec<u8> {
+        let mut encoded_key = Vec::new();
+        if counter.id().is_some() {
+            let key: IdCounterKey = counter.into();
+            encoded_key = postcard::to_extend(&1u8, encoded_key).unwrap();
+            encoded_key = postcard::to_extend(&key, encoded_key).unwrap();
+        } else {
+            let key: CounterKey = counter.into();
+            encoded_key = postcard::to_extend(&2u8, encoded_key).unwrap();
+            encoded_key = postcard::to_extend(&key, encoded_key).unwrap()
+        }
```

Comment:
A hard choice was made here... we prefix the encoding with 0x01 or 0x02 so that a future decoder knows which type of encoding was used.  The problem is that this makes it incompatible with the existing `key_for_counter` encoding.    I guess we could commit the `encoded_key = postcard::to_extend(&2u8, encoded_key).unwrap();` line to remain compatible, but then the encoding is non-determinist.  You could maybe even figure out how to craft an id so that it's encoding matches an existing key in the `key_for_counter` format.